mobile/src/bundles/common/components/button/button.tsx

```diff
@@ -1,19 +1,116 @@
-import React from 'react';
-import { Pressable } from 'react-native';
+import React, { useMemo } from 'react';
+import {
+    type PressableProps,
+    type StyleProp,
+    type TextStyle,
+    type ViewStyle,
+} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
 
-import { Text } from '~/bundles/common/components/components';
+import { Pressable, Text } from '~/bundles/common/components/components';
+import { globalStyles } from '~/bundles/common/styles/styles';
 
+import { ButtonType, TextCategory } from '../../enums/enums';
 import { styles } from './styles';
 
+type StylesProperties = {
+    basic: StyleProp<ViewStyle>;
+    pressed: StyleProp<ViewStyle | TextStyle>;
+    disabled: StyleProp<ViewStyle | TextStyle>;
+    label: StyleProp<ViewStyle | TextStyle>;
+};
+
+type ButtonName = (typeof ButtonType)[keyof typeof ButtonType];
+
 type Properties = {
     label: string;
-    onPress: () => void;
-};
+    buttonType?: ButtonName;
+    iconName?: string;
+    iconSize?: number;
+} & PressableProps;
+
+const iconDefaultSize = 32;
 
-const Button: React.FC<Properties> = ({ label, onPress }) => {
+const Button: React.FC<Properties> = ({
+    label,
+    style: pressableStyle,
+    iconName,
+    iconSize = iconDefaultSize,
+    buttonType = ButtonType.FILLED,
+    disabled = false,
+    ...props
+}) => {
+    const componentStyles: Record<ButtonName, StylesProperties> =
+        useMemo(() => {
+            return {
+                [ButtonType.FILLED]: {
+                    basic: styles.button_filled,
+                    pressed: styles.button_filled_pressed,
+                    disabled: styles.button_filled_disabled,
+                    label: styles.label,
+                },
+                [ButtonType.OUTLINE]: {
+                    basic: styles.button_outline,
+                    pressed: styles.button_outline_pressed,
+                    disabled: styles.button_outline_disabled,
+                    label: styles.content_pressed,
+                },
+                [ButtonType.GHOST]: {
+                    basic: styles.button_ghost,
+                    pressed: styles.button_ghost_pressed,
+                    disabled: styles.content_disabled,
+                    label: styles.content_pressed,
+                },
+            };
+        }, []);
+    const isFilledButton = ButtonType.FILLED === (buttonType as string);
```

Comment:
```suggestion
    const isFilledButton = ButtonType.FILLED === buttonType
```
I don't get any linter errors

Reply:
Strange thing, I will recheck configuration tomorrow, refactor code like it was before